components/operations-controller/main.go

```diff
@@ -95,23 +93,16 @@ func main() {
 		os.Exit(1)
 	}
 
-	httpClient := http.Client{
-		Transport: httputil.NewHTTPTransport(cfg.HttpClient),
-		Timeout:   cfg.HttpClient.Timeout,
-	}
-
-	unsignedTokenProvider, err := director.NewUnsignedTokenProvider(cfg.GraphQLClient.GraphqlEndpoint)
-	if err != nil {
-		fatalOnError(err)
-	}
+	httpClient, err := prepareHttpClient(cfg.HttpClient, cfg.GraphQLClient)
+	fatalOnError(err)
 
-	directorGraphQLClient, err := graphql.PrepareGqlClient(cfg.GraphQLClient, cfg.HttpClient, unsignedTokenProvider)
+	graphqlClient, err := graphql.PrepareGqlClientWithHttpClient(cfg.GraphQLClient, httpClient)
 	fatalOnError(err)
 
 	controller := controllers.NewOperationReconciler(cfg.Webhook, ctrl.Log.WithName("controllers").WithName("Operation"),
 		k8s.NewClient(mgr.GetClient()),
-		director.NewClient(cfg.Director.InternalAddress, httpClient, directorGraphQLClient),
-		webhook.NewClient(httpClient, defaultOAuthClientProviderFunc))
+		director.NewClient(cfg.Director.InternalAddress, httpClient, graphqlClient),
```

Comment:
you could create the graphql client within the director client constructor maybe ? Pass the graphql client config as part of the director client config